uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/OidcUserInfoController.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Licensed to Apereo under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright ownership. Apereo
+ * licenses this file to you under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the License at the
+ * following location:
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apereo.portal.rest;
+
+import javax.servlet.http.HttpServletRequest;
+import org.apereo.portal.security.IPerson;
+import org.apereo.portal.security.IPersonManager;
+import org.apereo.portal.security.oauth.IdTokenFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller provides an endpoint through which clients can obtain an ID Token in a format
+ * that is aligned with with OpenID Connect (OIDC). <i>Clients</i> in this case are normally content
+ * objects embedded within the portal page itself (e.g. portlets, soffits, or JS bundles).
+ *
+ * <p>It is critical to point out that this endpoint is not a compliant OIDC Identity Provider. It
+ * does not implement OAuth Authentication Flows of any sort. The user (<code>sub</code> in the ID
+ * Token) is either logged into the portal (ahead of time) or isn't.
+ *
+ * <p>The value of this endpoint is not (therefore) that it brings support for OIDC to uPortal, but
+ * that other modules and services designed to work with uPortal can implement security using
+ * standard OIDC approaches.
+ */
+@RestController
+public class OidcUserInfoController {
+
+    public static final String ENDPOINT_URI = "/v5-1/userinfo";
+    public static final String CONTENT_TYPE = "application/json"; // Should be application/jwt?
+
```

Comment:
Yes, make it application/jwt since it always returns a signed JWT:

> If the UserInfo Response is signed and/or encrypted, then the Claims are returned in a JWT and the content-type MUST be application/jwt.

Cf http://openid.net/specs/openid-connect-core-1_0.html#UserInfo

Reply:
Ah yes -- thanks for bringing that point up.

I tried `application/jwt` early on and couldn't get it to work in either Postman or the address bar.  I kept getting HTTP 406 Not Acceptable.  (I even tried setting an Accept header for application/jwt... no luck.)